### PR TITLE
[dev-menu] add label for disabled devsettings

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

We currently disable dev settings buttons that are not available for a given bundle but offer no explanation or indicator why they are disabled. 

This PR adds a message below the dev settings button when one or more options are unavailable

# How

Added a check for available dev setting features and a label below the buttons in the case that one or more options is unavailable

# Test Plan

Load an update and confirm that the label appears (the dev settings should be disabled on updates as they are not a local packager)

Ref: 

Screen Shot 2022-04-07 at 11 29 19 AM

# Checklist

- [x]  Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by ajsmth** [comment]
if you have any ideas for a better message / more clear language, feel free to suggest - I had some difficulty thinking of the best way to communicate this

File: packages/expo-dev-menu/app/components/Main.tsx [hunk 1]
```diff
@@ -188,6 +202,15 @@ export function Main() {
         </View>
       </View>
 
+      {!hasDisabledDevSettingOption && (
+        <>
+          <Spacer.Vertical size="large" />
+          <Text size="small" color="secondary" align="center">
+            Some development settings are unavailable for this build.
```
Comment (lukmccall) [inline packages/expo-dev-menu/app/components/Main.tsx:209]:
```suggestion
            Some development settings are unavailable for this development build.
```
I would add `development` to this message to indicate that we're describing the native build of the application. What do you think?

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-menu/app/components/Main.tsx (hunk 1 above)
Comment (ajsmth) [reply packages/expo-dev-menu/app/components/Main.tsx:209]: love it! thanks

File: packages/expo-dev-menu/app/components/Main.tsx [hunk 2]
```diff
@@ -188,6 +202,15 @@ export function Main() {
         </View>
       </View>
 
+      {!hasDisabledDevSettingOption && (
+        <>
+          <Spacer.Vertical size="large" />
+          <Text size="small" color="secondary" align="center">
+            Some development settings are unavailable for this development build.
```
Comment (ajsmth) [inline packages/expo-dev-menu/app/components/Main.tsx:209]:
```suggestion
            Some settings are unavailable for this development build.
```

**Pull request merged**